### PR TITLE
feat: asset install boilerplate

### PR DESCRIPTION
**Pull request by helording**
Boilerplates from the former asset install function. Temporarily put into a new folder to separate them from existing boilerplates. This is done so problematic files are easier to locate.  

- asset_install_boilerplate/assets to be pasted in assets folder of theme
- asset_install_boilerplate/snippets to be pasted in snippets folder of theme

- asset_install_boilerplate/locales/en.txt to be pasted at the bottom of locales/en.default.json
- asset_install_boilerplate/locales/jp.txt to be pasted at the bottom of locales/ja.json line

- asset_install_boilerplate/layout/theme_inserts.txt to be pasted into specified layout/theme.liquid lines

### CODE REVIEW

File: asset_install_boilerplates/layout/theme_inserts.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+== at line 237 ==
```
Comment (tieuu) [inline asset_install_boilerplates/layout/theme_inserts.txt:1]: is the line gonna be the same every time?

File: asset_install_boilerplates/layout/theme_inserts.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  {%- comment -%} END EASYPOINTS {%- endcomment -%}
+
+== at line 304 ==
+
+{%- comment -%} START EASYPOINTS - DO NOT MODIFY! 変更しないでください! INFO @ https://bit.ly/2Dn7ESM {%- endcomment -%}
+      {%- unless shop.metafields.loyalty.easy_points_attributes.value.stealth_mode -%}
+    {% render 'easy_points' %}
+  {%- endunless -%}
+
+  {%- comment -%} END EASYPOINTS {%- endcomment -%}
```
Comment (tieuu) [inline asset_install_boilerplates/layout/theme_inserts.txt:20]: also the indentation is weird man

File: asset_install_boilerplates/locales/en.txt
Comment (tieuu) [inline asset_install_boilerplates/locales/en.txt:1]: is there a reason why this is not a json file and is txt instead?

File: asset_install_boilerplates/snippets/easy_points.liquid [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            percent: null,
+            value: {{ shop_ep_metafield.reward_point_value }},
+            currency: {{ shop_ep_metafield.reward_currency_value }}
+          }
+        }
+      },
+      {%- if shop.customer_accounts_enabled and customer -%}
+        customer: {
+          id: {{ customer.id }},
+          hash: "{{ customer.id | append: shop.permanent_domain | append: "d96886fc8fb45617476366d9bddfe536" | md5 }}"
```
Comment (tieuu) [inline asset_install_boilerplates/snippets/easy_points.liquid:42]: iirc lorenzo said we dont need this?

File: asset_install_boilerplates/snippets/easy_points_bonus.liquid [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        assign current_bonus_points_active = true
+      elsif key_value[0] == 'currency_value'
+        assign currency_value = key_value[1]
+      elsif key_value[0] == 'point_value'
+        assign point_value = key_value[1] | times: 100000
+      endif
+
+      if current_bonus_points_active and currency_value > 0
+        assign point_ratio = point_value | divided_by: currency_value
+  	    if point_ratio > final_point_ratio
```
Comment (tieuu) [inline asset_install_boilerplates/snippets/easy_points_bonus.liquid:24]: dude the indentation is really weird

File: asset_install_boilerplates/snippets/easy_points_bonus.liquid [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+          assign final_currency_value = currency_value
+          assign final_point_value = point_value
+          assign final_point_ratio = final_point_value | divided_by: final_currency_value
+        endif
+  	  endif
```
Comment (tieuu) [inline asset_install_boilerplates/snippets/easy_points_bonus.liquid:29]: this one should be wrapping the smaller endif above too right?

File: asset_install_boilerplates/layout/theme_inserts.txt (hunk 1 above)
Comment (NexPB) [reply asset_install_boilerplates/layout/theme_inserts.txt:1]: No lel

File: asset_install_boilerplates/locales/en.txt
Comment (NexPB) [reply asset_install_boilerplates/locales/en.txt:1]: Yea should be `.json`

File: asset_install_boilerplates/snippets/easy_points_bonus.liquid (hunk 4 above)
Comment (helording) [reply asset_install_boilerplates/snippets/easy_points_bonus.liquid]: Seems to be fixed

File: asset_install_boilerplates/snippets/easy_points.liquid (hunk 3 above)
Comment (helording) [reply asset_install_boilerplates/snippets/easy_points.liquid]: Removed

File: asset_install_boilerplates/snippets/easy_points_bonus.liquid (hunk 5 above)
Comment (helording) [reply asset_install_boilerplates/snippets/easy_points_bonus.liquid]: Seems to be fixed

File: templates/jp.json
Comment (NexPB) [inline templates/jp.json:1]: Its `ja.json`and the folder is `locales/` - apply same to `en.json`

File: templates/jp.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        "hide_data": "Hide tier data",
+        "maintain_rank": "Maintain Rank",
+        "maintain_amount": "Amount",
+        "maintain_deadline": "Deadline",
+        "increase_rank": "Increase Rank",
+        "advancement_amount": "Amount",
+        "advancement_deadline": "Deadline"
+      }
+    }
+}
```
Comment (NexPB) [inline templates/jp.json:58]: End line break at end of files of your files

**Comment by azdevc** [comment]
> @azdevc can you make a separate pr for your intergration changes! This is not a PR meant for that.

File: snippets/easy_points.liquid [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    <input type="hidden" id="shopDomain" name="shopDomain" value="{{ shop.domain }}">
+    <input type="hidden" data-loyal-target="shop-point-rule-percent">
+    <input type="hidden" data-loyal-target="shop-point-rule-point-value" value="{{ shop_ep_metafield.point_value }}">
+    <input type="hidden" data-loyal-target="shop-point-rule-currency-value" value="{{ shop_ep_metafield.currency_value }}">
+    <input type="hidden" data-loyal-target="shop-reward-percent">
+    <input type="hidden" data-loyal-target="shop-reward-point-value" value="{{ shop_ep_metafield.reward_point_value }}">
+    <input type="hidden" data-loyal-target="shop-reward-currency-value" value="{{ shop_ep_metafield.reward_currency_value }}">
+    {%- if shop.customer_accounts_enabled and customer -%}
+      <input type="hidden" id="customerId" name="customerId" value="{{ customer.id }}">
+      <input type="hidden" id="customerHash" name="customerHash" value="{{ customer.id | append: shop.permanent_domain | append: "d96886fc8fb45617476366d9bddfe536" | md5 }}">
```
Comment (NexPB) [inline snippets/easy_points.liquid]: Still using hash.

File: assets/easy_points.js [hunk 8]
```diff
@@ -0,0 +1,950 @@
...
+      if (customerHashInput) {
+        clonedCustomerHashInput = customerHashInput.cloneNode();
+        clonedCustomerHashInput.removeAttribute("id");
+      }
+    }
+
+    if (clonedCustomerHashInput) {
+      clonedCustomerHashInput.name = name + '[customer_hash]';
+      form.appendChild(clonedCustomerHashInput);
+    }
```
Comment (NexPB) [inline assets/easy_points.js]: Code pbsolete you'll need to test if everything still works when removing.

**Review by NexPB — CHANGES_REQUESTED** [review]
Make sure to remove any obsolete hash core referrences.

File: templates/jp.json
Comment (NexPB) [reply templates/jp.json:1]: As discussed with Azino in discord lets name then `ja-ep.json` in the `/locales` folder

File: asset_install_boilerplates/layout/theme_inserts.txt (hunk 2 above)
Comment (helording) [reply asset_install_boilerplates/layout/theme_inserts.txt]: fixed

File: asset_install_boilerplates/layout/theme_inserts.txt (hunk 1 above)
Comment (helording) [reply asset_install_boilerplates/layout/theme_inserts.txt]: Ill just include the snippets then. Also changed the file to liquid for VSCode styling. Makes it easier to read.

File: asset_install_boilerplates/locales/en.txt
Comment (helording) [reply asset_install_boilerplates/locales/en.txt:1]: fixed

File: asset_install_boilerplates/snippets/easy_points_bonus.liquid (hunk 4 above)
Comment (helording) [reply asset_install_boilerplates/snippets/easy_points_bonus.liquid]: image

In my VSCode it looks like this ?_?... strange

File: templates/jp.json
Comment (helording) [reply templates/jp.json:1]: doneskies

File: templates/jp.json (hunk 6 above)
Comment (helording) [reply templates/jp.json:58]: fixed

File: assets/easy_points.js (hunk 8 above)
Comment (helording) [reply assets/easy_points.js:131]: Ok. For the meantime should we just comment it out along with hash codes so its easier to add back if needed?

Comment (NexPB) [reply assets/easy_points.js]: > Ok. For the meantime should we just comment it out along with hash codes so its easier to add back if needed?

Or we can check the git history if we need it after all :D

File: templates/jp.json
Comment (NexPB) [reply templates/jp.json:1]: `-ep` stands for easyPoints I didn't typo they should remain 2 separate files.

File: assets/easy_points.js (hunk 8 above)
Comment (helording) [reply assets/easy_points.js]: you right fam

File: locales/ja-en.json
Comment (NexPB) [inline locales/ja-en.json:1]: You probably didn't see my comment in the thread but please rename to:

- `en-ep.json`
- `ja-ep.json`

File: snippets/easy_points_total_points.liquid [hunk 9]
```diff
@@ -11,4 +11,4 @@
       </span>
     </span>
   {%- endunless -%}
-{%- comment -%} END EASYPOINTS {%- endcomment -%}
+{%- comment -%} END EASYPOINTS {%- endcomment -%}
```
Comment (NexPB) [inline snippets/easy_points_total_points.liquid:14]: You can change your vscode settings to inserts a linebreak at the end of your files : )

File: snippets/redemption_form.liquid
Comment (NexPB) [inline snippets/redemption_form.liquid:1]: We can rename these file to prepend `easy_points` while we're at it.

File: assets/easy_points.js
Comment (NexPB) [inline assets/easy_points.js:1]: Has more obsolute hash references :)

**Review by NexPB — CHANGES_REQUESTED** [review]

**Comment by NexPB** [comment]
@Gazwai once this is merged we should do the same as we did in solaris and replace the hardcoded multiplier values.

**Comment by Gazwai** [comment]
> replace the hardcoded multiplier values

Made a Linear on the EP workspace
https://linear.app/easypoints/issue/COR-74/asset-boilerplate-replace-hardcoded-multiplier-values

File: assets/easy_points.js
Comment (helording) [reply assets/easy_points.js:1]: I think I removed the values correctly but it might need to be doubled checked by

File: assets/easy_points.js [hunk 10]
```diff
@@ -693,18 +691,16 @@ window.addEventListener('DOMContentLoaded', function() {
 
     if (shopDomain && (
       force ||
-      (custId && custHash && staleSessionKey('customerMetafieldUpdatedAt', easyPointsSession)) ||
```
Comment (NexPB) [inline assets/easy_points.js]: Only the `custHash` should be removed.

File: assets/easy_points.js [hunk 11]
```diff
@@ -693,18 +691,16 @@ window.addEventListener('DOMContentLoaded', function() {
...
       var route = "/apps/loyalty/order_point_rule";
 
-      if (custId && custHash) {
-        data = { hash: custHash };
+      if (custId) {
         route = route + "/" + custId;
       }
 
-      var params = new URLSearchParams(data);
+      var params = new URLSearchParams();
```
Comment (NexPB) [inline assets/easy_points.js:703]: Seems like params is obsolete now.

File: assets/easy_points.js [hunk 12]
```diff
@@ -722,7 +718,7 @@ window.addEventListener('DOMContentLoaded', function() {
             easyPointsSession.rankAdvancementData = resp.tier_maintenance_data.advancement_data;
           }
 
-          if (!(custId && custHash)) {
+          if (!(custId)) {
```
Comment (NexPB) [inline assets/easy_points.js:721]: Remove extra `()` wrap

**Review by NexPB — CHANGES_REQUESTED** [review]

**Review by NexPB — APPROVED** [review]
Looks good to me now. Let's do one more integration based on this PR and then merge it in. Afterwards I'll check your Azino's PR.

**Pull request merged**